apps/tangle-dapp/utils/polkadot/identity.ts

```diff
@@ -40,37 +43,54 @@ export const extractDataFromIdentityInfo = (
   return null;
 };
 
+export const extractIdentityInfo = (
+  identityRegistration: PalletIdentityRegistration,
+): IdentityType => {
+  const info = identityRegistration.info;
+
+  const name = extractDataFromIdentityInfo(info, IdentityDataType.NAME);
+  const email = extractDataFromIdentityInfo(info, IdentityDataType.EMAIL);
+  const web = extractDataFromIdentityInfo(info, IdentityDataType.WEB);
+
+  const twitterName = extractDataFromIdentityInfo(
+    info,
+    IdentityDataType.TWITTER,
+  );
+
+  const twitter =
+    twitterName === null ? null : `https://x.com/${twitterName.substring(1)}`;
+
+  return {
+    name,
+    email,
+    web,
+    twitter,
+  } satisfies IdentityType;
+};
+
 export async function getAccountInfo(rpcEndpoint: string, address: string) {
   const api = await getApiPromise(rpcEndpoint);
   const identityData = await api.query.identity.identityOf(address);
 
-  if (identityData.isSome) {
-    const identity = identityData.unwrap();
-    const info = identity[0]?.info;
-
-    if (info) {
-      const name = extractDataFromIdentityInfo(info, IdentityDataType.NAME);
-      const email = extractDataFromIdentityInfo(info, IdentityDataType.EMAIL);
-      const web = extractDataFromIdentityInfo(info, IdentityDataType.WEB);
-
-      const twitterName = extractDataFromIdentityInfo(
-        info,
-        IdentityDataType.TWITTER,
-      );
-
-      const twitter =
-        twitterName === null
-          ? null
-          : `https://twitter.com/${twitterName.substring(1)}`;
-
-      return {
-        name,
-        email,
-        web,
-        twitter,
-      } satisfies IdentityType;
-    }
-  }
+  if (identityData.isNone) return null;
 
-  return null;
+  const [identityRegistration] = identityData.unwrap();
+
+  return extractIdentityInfo(identityRegistration);
+}
+
+export async function getMultipleAccountInfo(
+  rpcEndpoint: string,
+  addresses: string[],
+) {
+  const api = await getApiPromise(rpcEndpoint);
+  const identityData = await api.query.identity.identityOf.multi(addresses);
+
+  return identityData.map((data) => {
+    if (data.isNone) return null;
+
+    const [registration] = data.unwrap();
+
+    return extractIdentityInfo(registration);
+  });
```

Comment:
`.flatMap` and `if (data.isNone) return []` else `... return [extractIdentityInfo(registration)]` instead? Otherwise the consumer may need to manually filter out / ignore `null` values

Reply:
I need to add documentation for this function. Its purpose is to return an array of `IdentityType` or `null`, where each element corresponds to an address in the input array. If an address lacks identity information, the corresponding element will be null.